ecs-init/cache/cache.go

```diff
@@ -36,28 +39,37 @@ const (
 
 // Downloader is responsible for cache operations relating to downloading the agent
 type Downloader struct {
-	getter   httpGetter
-	fs       fileSystem
-	metadata instanceMetadata
-	region   string
+	s3downloader s3Downloader
+	fs           fileSystem
+	metadata     instanceMetadata
+	region       string
 }
 
 // NewDownloader returns a Downloader with default dependencies
-func NewDownloader() *Downloader {
+func NewDownloader() (*Downloader, error) {
 	downloader := &Downloader{
-		getter: customGetter,
-		fs:     &standardFS{},
+		fs: &standardFS{},
 	}
 
 	// If metadata cannot be initialized the region string is populated with the default value to prevent future
 	// calls to retrieve the region from metadata
 	sessionInstance, err := session.NewSession()
 	if err != nil {
-		downloader.region = config.DefaultRegionName
-		return downloader
+		return nil, err
```

Comment:
Please correct me if I read it wrong. The sessionInstance is only used to get the region, while if it failed it should use the default region. But with this change, it's going to fail the download process?

Reply:
it seems like so, but so is the original code... do you think we should ignore the error and just use the default region

Reply:
@richardpen is correct. that's a behavior change. let's maintain the same behavior as before with this PR. For now, logging the error and stating that default region is being used is good enough, while not failing the download itself.

Reply:
i was wrong above.. now i understand. will update the code